app/frontend/src/components/RatingsSlider/RatingsSlider.tsx

```diff
@@ -91,16 +89,15 @@ export default function RatingsSlider({ defaultValue, onChange }: SliderProps) {
         mark: classes.mark,
       }}
       aria-label={RATINGS_SLIDER}
-      value={currentValue}
+      value={value}
       min={0}
       max={1}
       step={0.01}
       marks={marks}
       valueLabelDisplay="on"
       valueLabelFormat={(value) => <SliderLabel value={value} />}
       onChange={(event, value) => {
-        typeof value === "number" && setCurrentValue(value);
-        onChange?.(currentValue);
```

Comment:
I think this was probably why it's one value behind - probably should be `onChange(value)` instead? But looks like you got something simpler working here and letting React Hook form manage state seems like a better idea as it manages re-renders more cleverly.

Reply:
Does setting state not set it immediately and mark the component as dirty?

Reply:
Nope, the state value doesn't change for that closure/particular render the component is in, so would still be referring to the old value. Also the state setting is batched by React so even if it does update, I guess it wouldn't update fast enough for the `onChange` here to read the correct value.